app/scripts/views/settings/clients.js

```diff
@@ -115,41 +136,52 @@ define(function (require, exports, module) {
       });
     },
 
-    _onDeviceAdded: function () {
+    _onItemAdded: function () {
       this.render();
     },
 
-    _onDeviceRemoved: function (device) {
-      var id = device.get('id');
-      var self = this;
-      $('#' + id).slideUp(DEVICE_REMOVED_ANIMATION_MS, function () {
-        // re-render in case the last device is removed and the
-        // "no registered devices" message needs to be shown.
-        self.render();
-      });
+    _onItemRemoved: function (item) {
+      $('#' + item.get('id')).slideUp(DEVICE_REMOVED_ANIMATION_MS);
     },
 
-    _onDisconnectDevice: function (event) {
-      this.logViewEvent('disconnect');
-      var deviceId = $(event.currentTarget).attr('data-id');
-      this._destroyDevice(deviceId);
+    _onDisconnectClient: function (event) {
+      var itemId = $(event.currentTarget).data('id');
+      // type of client that was disconnected, can be 'client' or 'device'.
+      var clientType = $(event.currentTarget).data('type');
+
+      this.logViewEvent(clientType + '.disconnect');
+      if (clientType === 'device') {
```

Comment:
I'm undecided in this case, but have this nagging feeling that our views have taken on too much control and are becoming unruly messes.

To help the view stay thin, you could possibly push this branching logic down into the AppsDevicesCollection and keep a long lived reference to an instance:

```js
this._appsAndDevicesCollection.removeService(itemId, account);
```

`removeService` would then lookup the type of the item and call the appropriate function.

I'm not sure how that'd pan out in reality.